app/controllers/spree/api/retailops/settlement_controller.rb

```diff
@@ -62,14 +62,30 @@ def mark_complete
           render text: @settlement_results.to_json
         end
 
+        # This gets called by RetailOps after a returned item has been
+        # received and accepted at the warehouse. We need to look elsehwere
+        # for hook when returned item has been received but can not be restocked.
+        #
+        # We first look for a `RetailopsReimbursementHandler` class to see if the host
+        # application has defined any custom logic. If they have, we simply
+        # pass the parameters to the defined `Reimbursement.from_retailops_add_refund`
+        # factory method and call `RetailopsReimbursementHandler#create_reimbursement`.
         def add_refund
-          ActiveRecord::Base.transaction do
-            find_order
-            assert_return params
-            @order.update!
+          if defined?(RetailopsReimbursementHandler)
+            reimbursement = RetailopsReimbursementHandler.from_retailops_add_refund(
+              params['settlement']
```

Comment:
Looks like it might just be a parameter that is not always sent then. The below example was pulled from our logs yesterday at 2016-03-23 21:28:30.908217+00:00. These logs are generated by Rails from the parameters sent to the application after parsing by the middleware; we did not add custom middleware. It looks like it's the same data duplicated inside the "settlement" parameter.

Is there anything in the RetailOps system that would do this before the request is sent? I would like to understand this better and spend time looking through parts of Spree and Rails that might be doing this if there is nothing on your side that could be doing this.

```json
{
  "shipping_amt": 0,
  "subtotal_amt": 74.98,
  "options": {},
  "product_amt": 74.98,
  "refund_amt": "81.54",
  "order_refnum": "R000000000",
  "tax_amt": 6.56,
  "rma_id": "20315",
  "return_items": [
    {
      "order_item_id": "557438",
      "channel_refnum": "856653",
      "sku": "134396",
      "quantity": "1"
    }
  ],
  "return_id": "19946",
  "refund_items": [
    {
      "refnum": "856653",
      "amount": "81.54",
      "quantity": "1",
      "id": "856653",
      "label": "Credit (return_item 24086)"
    }
  ],
  "settlement": {
    "shipping_amt": 0,
    "subtotal_amt": 74.98,
    "options": {},
    "product_amt": 74.98,
    "refund_amt": "81.54",
    "order_refnum": "R000000000",
    "tax_amt": 6.56,
    "rma_id": "20315",
    "return_items": [
      {
        "order_item_id": "557438",
        "channel_refnum": "856653",
        "sku": "134396",
        "quantity": "1"
      }
    ],
    "return_id": "19946",
    "refund_items": [
      {
        "refnum": "856653",
        "amount": "81.54",
        "quantity": "1",
        "id": "856653",
        "label": "Credit (return_item 24086)"
      }
    ]
  }
}
```